app/models/service/dialog_properties/retirement.rb

```diff
@@ -51,7 +48,10 @@ def retire_on_date=(value)
 
       def retirement_warning
         warn_value = parse_retirement_warn
-        @attributes[:retirement_warn] = warn_value if warn_value
+        if warn_value
+          days_between_warn_and_retirement = (@attributes[:retires_on] - warn_value) / 1.day
```

Comment:
@d-m-u Did you have thoughts on adding `to_i` to the first part of the equation as described in https://github.com/ManageIQ/manageiq/pull/18637#pullrequestreview-228852424 ?